### ISSUE TITLE
Machine bebop is incorrectly identified as anvil

### ISSUE THREAD

**Issue by dqwu** [E3SM-Project/E3SM#5485]
Steps to reproduce on bebop login node (e.g. hostname beboplogin3):
```
ssh bebop.lcrc.anl.gov

git clone https://github.com/E3SM-Project/E3SM.git
cd E3SM

git submodule update --init

cd cime/scripts
./create_newcase --case X_f19_g16 --compset X --res f19_g16
```
Output:
```
Compset longname is 2000_XATM_XLND_XICE_XOCN_XROF_XGLC_XWAV
...
Machine is anvil
```
NODENAME_REGEX to identify bebop is "beboplogin.\*"

This issue seems to be caused by https://github.com/E3SM-Project/E3SM/commit/321ad70e7023977ccc6f3bf0fb9a7ba9ca7b3dbc
NODENAME_REGEX to indentify anvil was changed from "blueslogin.\*.lcrc.anl.gov" to "b.\*.lcrc.anl.gov".
Since anvil is defined before bebop in config_machines.xml, node name beboplogin3.lcrc.anl.gov will be identified as anvil.

**Comment by dqwu** [comment]
@grnydawn Is it intentional to change "blueslogin.\*.lcrc.anl.gov" to "b.\*.lcrc.anl.gov" for anvil?

**Comment by sarich** [comment]
If you are running on an anvil compute node then the hostname is simply bXXX, I'm not sure if that is the reason for the change

**Comment by dqwu** [comment]
> If you are running on an anvil compute node then the hostname is simply bXXX, I'm not sure if that is the reason for the change

The issue is not on anvil, but on bebop. bXXX will also match beboplogin (use original bluesloginXXX will not).

**Comment by sarich** [comment]
Yes, but I don't want to just change it back to blueslogin* because then it will no longer resolve correctly on the anvil compute nodes

**Comment by sarich** [comment]
We could change it to (blueslogin.*|b\d+).*.lcrc.anl.gov

**Comment by dqwu** [comment]
> We could change it to (blueslogin._|b\d+)._.lcrc.anl.gov

As long as the updated regular expression for anvil can exclude bebop, it should be OK.
